public/Add-DbaReplArticle.ps1

```diff
@@ -0,0 +1,179 @@
+
+        Website: https://dbatools.io
+        Copyright: (c) 2023 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+        https://learn.microsoft.com/en-us/sql/relational-databases/replication/publish/define-an-article?view=sql-server-ver16#RMOProcedure
+
+    .LINK
+        https://dbatools.io/Add-DbaReplArticle
+
+    .EXAMPLE
+        PS C:\> Add-DbaReplArticle -SqlInstance mssql1 -Database Northwind -Publication PubFromPosh -Name TableToRepl
+
+        Adds the TableToRepl table to the PubFromPosh publication from mssql1.Northwind
+
+
+    .EXAMPLE
+        PS C:\> $article = @{
+                    SqlInstance           = "mssql1"
+                    Database              = "pubs"
+                    Publication           = "testPub"
+                    Name                  = "publishers"
+                    Filter                = "city = 'seattle'"
+                }
+        PS C:\> Add-DbaReplArticle @article -EnableException
+
+        Adds the publishers table to the TestPub publication from mssql1.Pubs with a horizontal filter of only rows where city = 'seattle.
+
+    .EXAMPLE
+        PS C:\> $cso = New-DbaReplCreationScriptOptions -Options NonClusteredIndexes, Statistics
+        PS C:\> $article = @{
+                    SqlInstance           = 'mssql1'
+                    Database              = 'pubs'
+                    Publication           = 'testPub'
+                    Name                  = 'stores'
+                    CreationScriptOptions = $cso
+                }
+        PS C:\> Add-DbaReplArticle @article -EnableException
+
+        Adds the stores table to the testPub publication from mssql1.pubs with the NonClusteredIndexes and Statistics options set
+        includes default options.
+    #>
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess, ConfirmImpact = 'Medium')]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [parameter(Mandatory)]
+        [String]$Database,
+        [parameter(Mandatory)]
+        [String]$Publication,
+        [String]$Schema = 'dbo',
+        [parameter(Mandatory)]
+        [String]$Name,
+        [String]$Filter,
+        [PSObject]$CreationScriptOptions,
+        [Switch]$EnableException
+    )
+    process {
+
+        # Check that $CreationScriptOptions is a valid object
+        if ($CreationScriptOptions -and ($CreationScriptOptions -isnot [Microsoft.SqlServer.Replication.CreationScriptOptions])) {
+            Stop-Function -Message "CreationScriptOptions should be the right type. Use New-DbaReplCreationScriptOptions to create the object" -ErrorRecord $_ -Target $instance -Continue
+        }
+
+        foreach ($instance in $SqlInstance) {
+            try {
+                $replServer = Get-DbaReplServer -SqlInstance $instance -SqlCredential $SqlCredential
```

Comment:
```suggestion
                $replServer = Get-DbaReplServer -SqlInstance $instance -SqlCredential $SqlCredential -EnableException:$EnableException
```

Any call to another public function should have the EnableException passed in so downstream will throw exceptions as well.

Reply:
Good point, thanks!